include/envoy/http/filter.h

```diff
@@ -283,6 +283,17 @@ class StreamFilterCallbacks {
    * @return the ScopeTrackedObject for this stream.
    */
   virtual const ScopeTrackedObject& scope() PURE;
+
+  /**
+   * Should be used when a filter's callback is executed, and we continue
+   * processing the request. Some crash context that is lost when we resume the
+   * filter chain from commonContinue(), can be restored with this.
+   *
+   * @return an opaque ScopeTrackedObject that will restore context when
+   * continuing.
+   */
+  // TODO(kbaichoo): better naming and commenting
```

Comment:
How is this different from the scope() method above?  It seems very similar.

Reply:
The one above just returns a ScopeTrackedObject of `this` while the idea behind the `restoreContextOnContinue` is to gather various `ScopeTrackedObjects` that the underlying object has access to and have those be encapsulated and registered to provide crash context.